src/pip/_internal/cli/main_parser.py

```diff
@@ -57,6 +102,27 @@ def parse_command(args: List[str]) -> Tuple[str, List[str]]:
     #  args_else: ['install', '--user', 'INITools']
     general_options, args_else = parser.parse_args(args)
 
+    # --python
+    if general_options.python and "_PIP_RUNNING_IN_SUBPROCESS" not in os.environ:
+        # Re-invoke pip using the specified Python interpreter
+        interpreter = identify_python_interpreter(general_options.python)
+        if interpreter is None:
+            raise CommandError(
+                f"Could not locate Python interpreter {general_options.python}"
+            )
+
+        pip_cmd = [
+            interpreter,
+            get_runnable_pip(),
+        ]
+        pip_cmd.extend(args)
+
+        # Set a flag so the child doesn't re-invoke itself, causing
+        # an infinite loop.
+        os.environ["_PIP_RUNNING_IN_SUBPROCESS"] = "1"
+        proc = subprocess.run(pip_cmd)
```

Comment:
Wrap this in a try/except to avoid printing a stack trace if the subprocess can't start for any reason ?

Reply:
Done. I'm not sure how to test this, though. A unit test (mocking `subprocess.run`) would basically just test whether the `try...except` statement works, and for a functional test, I'd need a (cross-platform) executable I could supply that would error, and that isn't already trapped by the various checks we do.

I'm going to not bother testing, but let me know if you can think of a way of adding a test 😉